### PR TITLE
Edit view

### PR DESCRIPTION
**Pull request by david-forster10**
Changes made:
- Created file for ModalEdit
- Modified ProjectForm for reusing the fields from creating a project for use when editing
- Modified ModalInput to pass a true value to ProjectForm so that all fields to create a project and request are shown.
- Modified ListItem so that when the "Edit" button is clicked, the edit view is displayed.

### CODE REVIEW

File: src/components/Shared/List/ListItem.jsx [hunk 1]
```diff
@@ -3,18 +3,21 @@
...
+import ModalEdit from '../ModalEdit/ModalEdit';
 import './css/ListItem.css';
 
 export default class ListItem extends Component {
 	state = {
-		open: false
+		open: false,
+		mpShow: false,
 	}
 	/*eslint-disable */
```
Comment (gwolverson) [inline src/components/Shared/List/ListItem.jsx]: Why is the linting being disabled here??

File: src/components/Shared/ModalEdit/ModalEdit.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import {Modal, Button, Row, Col, Label} from 'react-bootstrap';
+import ProjectForm from '../ProjectForm';
+
+export default class ModalEdit extends React.Component {
+	initialState = {
```
Comment (gwolverson) [inline src/components/Shared/ModalEdit/ModalEdit.jsx]: what's the purpose of initialState? Why aren't we just setting the value of `state` straight away? (Keep It Simple)

File: src/components/Shared/ProjectForm.jsx [hunk 3]
```diff
@@ -4,139 +4,193 @@ import { Row, Col, Form, FormGroup, ControlLabel, FormControl, HelpBlock} from '
...
+							</Col>
+						</Row>
+					</Col>
+				</Row>
+			</Col>
+		);
+	}
+}
+
+export class ProjectsLayout extends React.Component {
```
Comment (gwolverson) [inline src/components/Shared/ProjectForm.jsx]: This should be moved to its own file

File: src/components/Shared/ProjectForm.jsx [hunk 4]
```diff
@@ -4,139 +4,193 @@ import { Row, Col, Form, FormGroup, ControlLabel, FormControl, HelpBlock} from '
...
+			return (
+				<Row id='project-form'>
+					<ProjectsLayout handleProjectUpdate={this.handleFormUpdate}/>
+				</Row>
+			); 
+		}
+	}
+}
+	
+export class RequestsLayout extends React.Component {
```
Comment (gwolverson) [inline src/components/Shared/ProjectForm.jsx]: This should be moved to its own file

File: src/components/Shared/List/ListItem.jsx (hunk 1 above)
Comment (david-forster10) [reply src/components/Shared/List/ListItem.jsx]: Code not changed following discussions with Greg

File: src/components/Shared/ProjectForm.jsx (hunk 3 above)
Comment (david-forster10) [reply src/components/Shared/ProjectForm.jsx]: Done

File: src/components/Shared/ProjectForm.jsx (hunk 4 above)
Comment (david-forster10) [reply src/components/Shared/ProjectForm.jsx]: Done

File: src/components/Shared/ModalEdit/ModalEdit.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+export default class ModalEdit extends React.Component {
+	state = {
+		modalData: {},
+		saveBtnState: 'Save',
+		saveState: 'success',
+		actionDescription: ''
+	}
+
+	saveForm = () => {
```
Comment (adamkhazi) [inline src/components/Shared/ModalEdit/ModalEdit.jsx]: Can this function potentially be reused across ModalEdit and ModalInput given that only actionDescription is different? i.e. added to an external file and imported into ModalEdit and ModalInput. Would be nice later on if the timeouts need to be changed and they could be altered in a single location.

File: src/components/Shared/ViewsforModal/ProjectsLayout.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+					<FieldGroup id='projectStatus' onChange={this.handleFormUpdate} type='text' label='Status:' placeholder='' inputRef={ ref => {
+						this.projectStatus = ref;
+					}}/>
+				</Form>
+			</Col>
+		);
+	}
+}
+
+export class FieldGroup extends React.Component {
```
Comment (gwolverson) [inline src/components/Shared/ViewsforModal/ProjectsLayout.jsx]: Extract into its own file for re-use

**Pull request merged**